compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2986,7 +2986,18 @@ extends ReferenceMsg(CannotBeAccessedID):
         i"none of the overloaded alternatives named $name can"
     val where = if (ctx.owner.exists) i" from ${ctx.owner.enclosingClass}" else ""
     val whyNot = new StringBuffer
-    alts.foreach(_.isAccessibleFrom(pre, superAccess, whyNot))
+    for alt <- alts do
+      val cls = alt.owner.enclosingSubClass
+      val owner = if cls.exists then cls else alt.owner
+      val location =
+        if alt.is(Protected) && !owner.is(Final) then owner.showLocated  + " or one of its subclasses"
```

Comment:
there is still an extra subtlety here, e.g. for `protected[pkg]` in open class `Foo`, then its `Foo or one of its subclasses, or within pkg`